internal/controller/datadogagent/component/agent/default.go

```diff
@@ -665,10 +674,19 @@ func volumeMountsForInitConfig() []corev1.VolumeMount {
 }
 
 func volumesForAgent(dda metav1.Object, requiredContainers []apicommon.AgentContainerName) []corev1.Volume {
+	// Use the original DDA name for dependencies that are created by the DatadogAgent controller
+	ddaForDependencies := dda
+	if originalDDAName, ok := dda.GetLabels()[apicommon.DatadogAgentNameLabelKey]; ok {
+		ddaForDependencies = &metav1.ObjectMeta{
+			Name:      originalDDAName,
+			Namespace: dda.GetNamespace(),
+		}
+	}
+
```

Comment:
Same comment as above in case of `GetVolumeInstallInfo`. However, 
`GetVolumeForSecurity` uses a different [function](https://github.com/DataDog/datadog-operator/blob/8c52bc1fc73e15b8d0e45457b34068ba46e50910/internal/controller/datadogagent/common/utils.go#L89-L91) `GetDefaultSeccompConfigMapName` and gets the DDA name directly from metadata. Same function is used [here](https://github.com/DataDog/datadog-operator/blob/8c52bc1fc73e15b8d0e45457b34068ba46e50910/internal/controller/datadogagent/global/dependencies.go#L99-L107) too so I wonder if the fix should be in the `GetDefaultSeccompConfigMapName` so they use common name and this file may not need to change.

Reply:
Thanks, great call indeed, added in 38a11de03fc4d1f8894e7b2b7945804b1762fe9e